### ISSUE TITLE
Test the pretrained model with own imagesets

### ISSUE THREAD

**Issue by landkwon94** [MaybeShewill-CV/lanenet-lane-detection#532]
Hello Sir, I am a master course student in the department of Ecological Landscape Architecture from South Korea.
Most of all, thank you so much for your nice research and python scripts! I am really excited in your works :)

By the way, I have some question about source code.

------ 1st question ------
I used your pre-trained model which was trained with TuSimple datasets for testing my Own images.
I also converted a little bit of your 'lanenet_postprocess.py' because my images seem different with TuSimple datasets :)

And the images below are the results.

1) src image
![src_image](https://user-images.githubusercontent.com/73383617/144185321-25734d25-176b-4174-8d24-e52eabf7fa96.png)

2) mask image
![mask_image](https://user-images.githubusercontent.com/73383617/144185342-3bb92ff2-47a0-448d-8e54-18036093912b.png)

3) instance image
![instance_image](https://user-images.githubusercontent.com/73383617/144185365-b7144ec6-bdaf-490c-9f99-7ee0820eaca1.png)

4) binary image
![binary_image](https://user-images.githubusercontent.com/73383617/144185390-e489d7c7-44bf-4437-b834-ba4014e2b8e5.png)

As you see the image, we can find that the model is finding almost all lanes very well from the 'instance' and 'binary' images.

However, I am curious why there is only one lane instance from the 'src' and 'mask' images.

I read all 'github issues' at here, and I thought 'Is it because of remap.yml?'.
But I am not sure, because my image shows 'one' lane.
Most people who wrote issues about 'remap.yml' showed that their image showed nothing..

Or,, is there some codes which is about setting some parameters?
For example, some threshold..?

------ 2nd question ------
In addition, I have one question about making training datasets.
You used label.json files which were created by TuSimple datasets.
It seems like you converted those .json files to training_images from 'generate_tusimple_dataset.py'

My plan is to create my own training datasets directly to 'IMAGE', not json file.
Then, is it okay to do so?
And if so, do I have to apply the Same color with your source code?

![20211201_160936](https://user-images.githubusercontent.com/73383617/144188110-305b7514-d878-4e23-9a1f-ceb3aa005d46.png)

------ 3rd question ------
Also, I want to increase the maximum number of classes from 8 classes to eg. 10~12.
Then, is it okay to just simply change this part of the code?

Because I am not sure, whether it will affect other codes somehow..

![20211201_161115](https://user-images.githubusercontent.com/73383617/144188457-29d51a4b-4600-41c8-8df3-70f642d446cb.png)

Again, thank you so much for your nice codes!
These days, I am really excited to learn about your research :)

I will wait for your comments! Many thanks again, Sir.

From South Korea.

**Comment by landkwon94** [comment]
To leave one more comments about 3rd question, I have plan to increase the number of class.
LaneNet is detecting whether lane or not. But I have to identify 1) normal lane, 2) center lane, 3) edge lane, 4) background.

So I will change config file a little bit (number of class).

My question is that, then do I have to make different format of training datasets?
For example, you made 1) binary (0-background, 255-lane) / 2) instance (each lane) / 3) image.
Then is it right that I should make datasets 1) binary (0-background, 50-center lane, 100-lane, 200-edge) / 2) instance (each center-normal-edge lane) / 3) image??

I am not sure that my understanding is right..
I really need your comments, Sir!

Thank you for reading my long question :)

**Comment by landkwon94** [comment]
While I am running your training code, I got some issues..

![image](https://user-images.githubusercontent.com/73383617/144838258-70a2b3c6-5f83-443b-abaa-1bbbc79ff8bc.png)

I trained the model with your custom tuSimple data(which is in your data folder), also I changed batch size - 2 (Because there are only 6 images in the folder). But there is only NaN loss.....

Because I am not good at programming that much, it is so hard to debug from zero haha... May I ask you some help for me?

Thank you again ....

**Comment by landkwon94** [comment]
I solved all the issues by myself (including converting the code to multi-class detection), and it seems the developer of this code doesn't seem to help users :)
Anyway, thank you for nice code!

**Comment by MaybeShewill-CV** [comment]
@landkwon94 Sorry for the delay repley. I'm not a student in school and work's busy at the end of the year. Sorry again:)

**Comment by landkwon94** [comment]
@MaybeShewill-CV Ahh, you are not a student now! I understand your busy days :) Many thanks for your nice code again, I think I will use your paper for my thesis reference! Happy new year :)

**Comment by MaybeShewill-CV** [comment]
@landkwon94 That's not my paper. Only a re-implement of the origin paper. You may reference the origin paper:)

**Comment by hojunpark12** [comment]
> While I am running your training code, I got some issues..
> 
> ![image](https://user-images.githubusercontent.com/73383617/144838258-70a2b3c6-5f83-443b-abaa-1bbbc79ff8bc.png)
> 
> I trained the model with your custom tuSimple data(which is in your data folder), also I changed batch size - 2 (Because there are only 6 images in the folder). But there is only NaN loss.....
> 
> Because I am not good at programming that much, it is so hard to debug from zero haha... May I ask you some help for me?
> 
> Thank you again ....

hello i also undergo same issue, can you say how solve this problem. 
i also trained the model with your custom tuSimple data
혹시 실례가 안되면 어떻게 해결하셨는지 여쭤봐도 될까요??

**Comment by landkwon94** [comment]
@hojunpark12 저 이거 해보다가 코드가 너무 에러도 많고 옛날버전이라 파이토치 버전으로 했어요ㅋㅋㅋ 레인넷 파이토치로 구글링하시면 코드 몇개 뜨는데 그걸로 해보시는걸 추천해요. 대신에 remap은 파이토치 버전으로 하신 분들이 없으신점 참고바래요. 저는 차선 탐지만 하면돼서 그냥 그 부분은 접었습니다ㅋㅋㅋㅋ;; 더 궁금하신 점 있으시면 댓글 주세요~!

**Comment by hojunpark12** [comment]
> @hojunpark12 저 이거 해보다가 코드가 너무 에러도 많고 옛날버전이라 파이토치 버전으로 했어요ㅋㅋㅋ 레인넷 파이토치로 구글링하시면 코드 몇개 뜨는데 그걸로 해보시는걸 추천해요. 대신에 remap은 파이토치 버전으로 하신 분들이 없으신점 참고바래요. 저는 차선 탐지만 하면돼서 그냥 그 부분은 접었습니다ㅋㅋㅋㅋ;; 더 궁금하신 점 있으시면 댓글 주세요~!

헉.. 답장해주셔서 감사합니다. 다른 분들 오류 대부분 살펴보니까 다들 각자 코드 조금씩 바꿔가면서 해결하신것 같더라구요, 거의 반 포기 상태였는데 ㅜㅜㅜ 저도 차선 탐지만 하면 되서 레인넷 파이토치로 저도 해보려고 합니다. ㅜㅜ 정말 감사합니다.!!!!! 말 안해주셨으면 혼자 계속 삽질하고 있었을 것 같네요 정말 감사합니다!

**Comment by Gabrielle-Ribeiro** [comment]
Hello, @landkwon94! 

I'm having the same problem of your first question. With the src image and mask image missing some lanes. 
How did you solve this issue? 

Thank you.

**Comment by landkwon94** [comment]
@Gabrielle-Ribeiro 
I couldn't solve the issue, so I just change the model from this one to Pytorch LaneNet. You can find some cases from Google, it will help you a lot. But most of them doesn't include the post-processing part so I created my own DBSCAN code for this. If you have any more question,feel free to contact me :)

**Comment by gabaebae123** [comment]
> 헉.. 답장해주셔서 감사합니다. 다른 분들 오류 대부분 살펴보니까 다들 각자 코드 조금씩 바꿔가면서 해결하신것 같더라구요, 거의 반 포기 상태였는데 ㅜㅜㅜ 저도 차선 탐지만 하면 되서 레인넷 파이토치로 저도 해보려고 합니다. ㅜㅜ 정말 감사합니다.!!!!! 말 안해주셨으면 혼자 계속 삽질하고 있었을 것 같네요 정말 감사합니다!

Hello! Now I have also encountered the first problem you mentioned above, could you share your DBScan post-processing code? Looking forward to your reply！
